classes/controller/FrontController.php

```diff
@@ -675,12 +675,14 @@ protected function smartyOutputContent($content)
 
         $html = '';
 
+        $theme = $this->context->shop->theme->getName();
+
         if (is_array($content)) {
             foreach ($content as $tpl) {
-                $html .= $this->context->smarty->fetch($tpl, null, $this->getLayout());
+                $html .= $this->context->smarty->fetch($tpl, null, $theme . $this->getLayout());
```

Comment:
same here, maybe integrate shop id & lang_id in the compile_id parameter?

Reply:
It’s a compilation id not a cache id. See my previous response.